redux/reducers.js

```diff
@@ -0,0 +1,60 @@
+import { combineReducers } from 'redux';
+import * as types from "./actionTypes";
+
+/* Current shape of state:
+ {
+  user: "David",
+  currentChat: id 
+  chats: [ { id: 1,
+    name: 'Tea Chats',
+    users: ['David', 'Nick'] },
+    { id: 2,
+      name: 'Coffee Chats',
+      users: ['David', 'Ryan'] } ]
+  messages: [
+    { id: 0, chats_id: 1, user: 'David', message: 'ayyyyy', emojis: { David: ':joy:', Nick: ':laughing:' } },
+    { id: 1, chats_id: 1, user: 'Nick', message: 'lmao', emojis: {} }
+  ]
+} */
+
+const messageInitialState = {
+  messages: []
+};
+
+const userInitialState = {
+  user: '',
+  currentChat: 0,
+  chats: []
+};
+
+function messageReducer(state = messageInitialState, action) {  
+  console.log("messageReducer");
+  console.log(state); 
+    switch (action.type) {
+    case types.ADD_MESSAGE:
+      return {...state, messages: state.messages.concat(action.payload) };
```

Comment:
concat is used with 2 arrays, and i think the payload is going to come back as an object so you'll need to use a different method.

suggestion:
`const messages = [...state.messages, action.payload];   // merge the new message with the current message array
return {...state, messages}`

Reply:
oh, nm, i just tried it and you can concat an object. the js docs confused me since it says it has to be arrays? i usually do it in the way above so i had looked it up.

Reply:
I think the idea is to create a new object though instead of modifying the existing one, following the functional programming paradigm. So I would opt to go with @mnicole's method. Also, it would be good to check your formatting, @ernewtoner, e.g. return statements could be reformatted like this for better readability:
```
return {
   ...state,
   messages
}
```

Reply:
I'm not sure there's a difference between these methods, concat also just creates a new object. I almost did that formatting, it actually looks less readable to me for small return statements, makes these simple reducers look a lot more verbose, but yeah can update if that's the standard we want to follow.

Reply:
Nevermind on the formatting, I became convinced looking at the cleaner actions.js :laughing: